ethereum/contracts/ETHVault.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+// Holds ETH on behalf of ETHApp
+contract ETHVault is Ownable {
+    // Accepts ETH from the caller.
+    function lock() 
+        public
+        payable
+        onlyOwner
+    {
+        require(msg.value > 0, "Value of transaction must be positive");
+    }
+
+    // Returns ETH to the caller.
+    function unlock(uint128 _amount)
+        public
```

Comment:
Should use `external` instead of `public`, since we know that `unlock` won't ever be called internally (in the contract its defined). Using `external` saves a small amount of gas.

See details here: https://ethereum.stackexchange.com/questions/19380/external-vs-public-best-practices